lib/src/git_backend.rs

```diff
@@ -93,13 +95,16 @@ const NO_GC_REF_NAMESPACE: &str = "refs/jj/keep/";
 const CONFLICT_SUFFIX: &str = ".jjconflict";
 
 pub const JJ_TREES_COMMIT_HEADER: &[u8] = b"jj:trees";
+pub const CHANGE_ID_COMMIT_HEADER: &[u8] = b"change-id";
 
 #[derive(Debug, Error)]
 pub enum GitBackendInitError {
     #[error("Failed to initialize git repository")]
     InitRepository(#[source] gix::init::Error),
     #[error("Failed to open git repository")]
     OpenRepository(#[source] gix::open::Error),
+    #[error("Failed to open git backend due to missing config")]
+    MissingGitConfig(#[source] ConfigGetError),
```

Comment:
nit: I would make it `#[error(transparent)] Config(..)` or something. Since the default config should be valid, the error would likely be caused by bad user value.